/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
==============================================================================*/

syntax = "proto2";

package lstm_object_detection.tflite.protos;

import "protos/box_encodings.proto";

// DetectionResults is a list of Detection.
message DetectionResults {
  repeated Detection detection = 1;
}

// Detection consists of a bounding box, class confidences and indices.
message Detection {
  // Each detection message consists of only one bounding box.
  optional BoxCornerEncoding box = 1;
  // A box can be associated with multiple confidences for multiple classes.
  repeated float score = 2;
  repeated int32 class_index = 3;
  // Optional, for readability and easier access for external modules.
  // A unique name that identifies the class, e.g. a MID.
  repeated string class_name = 4;
  // A human readable name of the class.
  repeated string display_name = 5;
}
